src/cpp/modules/manualEigenVector/ManualEigenVectorGMM.cpp

```diff
@@ -0,0 +1,78 @@
+// ManualEigenVectorGMM.cpp : Defines the exported functions for the DLL.
+#include "ManualEigenVectorGMM.h"
+#include "../../shared/gmm.h"
+#include "gmm_vector_d.h"
+#include "memory_size.h"
+
+#include <iostream>
+#include <iomanip>
+#include <memory>
+
+// This function must be called before any other function.
+void ManualEigenVectorGMM::prepare(GMMInput&& input)
+{
+    _input = input;
+    int Jcols = (_input.k * (_input.d + 1) * (_input.d + 2)) / 2;
+    _output = { 0,  std::vector<double>(Jcols) };
+
+    // If it is not enough memory in system to store all data that
+    // manualEigenVector need, then executing will interrupt
+    // to prevent system deadlock
```

Comment:
If _there_ is not enough memory in _the_ system to store all _the_ data _structures_ that manualEigenVector _needs_, then _the execution_ will interrupt to prevent _the_ system _from deadlocking_